packages/data-stores/src/site/actions.ts

```diff
@@ -309,18 +310,33 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		 */
 		const anchorDesigns = [ 'hannah', 'gilbert', 'riley' ];
 		if ( anchorDesigns.indexOf( selectedDesign.template ) < 0 ) {
+			const themeSetupOptions: ThemeSetupOptions = {
+				trim_content: true,
+			};
+
+			if ( selectedDesign.verticalizable && siteVerticalId ) {
```

Comment:
```suggestion
			if ( selectedDesign.verticalizable ) {
```

and actually even the whole logic in this PR is not necessary anymore after D87177-code is deployed. The endpoint will consider vertical_id as null if it is passed empty string (a falsy value ini PHP). CMIIW.

Your call if we still want to keep this and the tests :)